### DISCUSSION TITLE
Always a flat trend arrow in Nightscout upload

### DISCUSSION THREAD

**Discussion by gpgerrard** [NightscoutFoundation/xDrip#2178] [General]
Using Xdrip (13 June nightly) getting readings from CamAPS FX. Xdrip shows the correct trend arrow but the MongoDB entries show that the upload is always 'Flat' regardless of the delta. Three consecutive readings below showing steep falls (and downward trend arrow in Xdrip) but all with 'Flat' direction in the upload. 

Thanks to all the devs for adding CamAPS support by the way - it's been a game changer for us and no longer have to have a second phone collecting readings in the pump slot so we can follow our 3yo daughter more easily!

`{  "_id": {    "$oid": "62a8e1d9c112d8e8512b8045"  },  "device": "xDrip-UiBased",  "date": 1655235030515,  "dateString": "2022-06-14T19:30:30.515Z",  "sgv": 85,  "delta": -9.923,  "direction": "Flat",  "type": "sgv",  "filtered": 0,  "unfiltered": 0,  "rssi": 100,  "noise": 1,  "sysTime": "2022-06-14T19:30:30.515Z",  "utcOffset": 60}`

`{  "_id": {    "$oid": "62a8e0abc112d8e85128b6bc"  },  "device": "xDrip-UiBased",  "date": 1655234728191,  "dateString": "2022-06-14T19:25:28.191Z",  "sgv": 95,  "delta": -10.991,  "direction": "Flat",  "type": "sgv",  "filtered": 0,  "unfiltered": 0,  "rssi": 100,  "noise": 1,  "sysTime": "2022-06-14T19:25:28.191Z",  "utcOffset": 60}`

`{  "_id": {    "$oid": "62a8df7fc112d8e85125de33"  },  "device": "xDrip-UiBased",  "date": 1655234427957,  "dateString": "2022-06-14T19:20:27.957Z",  "sgv": 106,  "delta": -15.053,  "direction": "Flat",  "type": "sgv",  "filtered": 0,  "unfiltered": 0,  "rssi": 100,  "noise": 1,  "sysTime": "2022-06-14T19:20:27.957Z",  "utcOffset": 60}`

**Comment by Navid200** [comment]
xDrip always calculates the trend arrow using the current and the previous reading.
But, other apps may take more than those two readings into account for calculating the trend.

This results in discrepancies.  It's not necessarily an issue.  

These are examples:
https://github.com/NightscoutFoundation/xDrip/issues/414
https://github.com/NightscoutFoundation/xDrip/issues/691
https://github.com/NightscoutFoundation/xDrip/issues/800
https://github.com/NightscoutFoundation/xDrip/issues/920

**Comment by tolot27** [comment]
Do you also upload to Nightscout at the same time? It is quite interesting that the `doMongoUpload` function [does not upload](https://github.com/NightscoutFoundation/xDrip/blob/b033ba4dfc404b9b5ade76436e0457ba2451061e/app/src/main/java/com/eveningoutpost/dexdrip/UtilityModels/NightscoutUploader.java#L1182) the delta value but it is shown in your records above. Where do you get these records from?

**Reply by Navid200** [reply]
I'm not 100% sure this is relevant.
But, when I look at the BG data table, I see data trends that are coming from My G6, which are identical to the trend arrow shown on my BlueJay, but different from the xDrip trend arrows.
Is it possible that xDrip uploads the entire data coming from G6 including the trend arrow values?

**Reply by gpgerrard** [reply]
@tolot27 The entries are from my Nightscout database (mycgm.entries) @Navid200  I appreciate there can be discrepancies in the trend arrow between apps but it's flat for each and every one of the 288 readings each day, I just picked three...

**Reply by Navid200** [reply]
Do you have any instances where your readings consistently, for more than 4 consecutive readings, trend low or trend high and preferably not at a low pace?
It is possible that you have a very good glucose control?  In that caser, if your fluctuations are minimal, G6 will only show flat trend arrows.

**Reply by Navid200** [reply]
Look at my data table.  My readings are consistently dropping.  yet, you can see the G6 trend is flat.
![Screenshot_20220614-174652](https://user-images.githubusercontent.com/51497406/173695062-b8267c5b-2748-4104-88b4-1e8049c6ef49.png)

**Reply by gpgerrard** [reply]
If only! She's 3 and we have good days and bad days. Today was a bad day with huge glycaemic variability and steep rises/falls due to a failed cannula. Flat arrow for every single reading in Nightscout but Xdrip showing appropriate arrows to reflect the trend.

**Reply by Navid200** [reply]
What does it show in the data table?

**Reply by gpgerrard** [reply]
Whereabouts is that in the app? Have had a look but can't find it! Thanks

**Reply by Navid200** [reply]
https://navid200.github.io/xDrip/docs/Datatables.html

**Reply by gpgerrard** [reply]
No arrow at all in the data table 
![Screenshot_20220614-225848_xDrip+](https://user-images.githubusercontent.com/52867075/173696408-6af71e02-1a9f-497d-bb63-1b0fbe82e8bd.jpg)

**Reply by jamorham** [reply]
I think I can resolve this with some work but I'll need to work closely with a camaps user like OP over multiple iterations. If you're interested to work it out, please can you email me at jamorham at gmail

**Comment by gpgerrard** [comment]
Will do, happy to help! Thanks again.
